diff --git a/node_modules/bmaplib.lushu/index.js b/node_modules/bmaplib.lushu/index.js
index a793df6..5478b9a 100644
--- a/node_modules/bmaplib.lushu/index.js
+++ b/node_modules/bmaplib.lushu/index.js
@@ -277,7 +277,9 @@
                 target_pos = this._projection.lngLatToPoint(targetPos),
                 count = Math.round(me._getDistance(init_pos, target_pos) / step);
             if (count < 1) {
-                me._moveNext(++me.i);
+                setTimeout(() => { // 解决lushu重复点过多 内存溢出
+                    me._moveNext(++me.i);
+                }, 0)
                 return;
             }
             me._intervalFlag = setInterval(function() {
