diff --git a/node_modules/bmaplib.lushu/index.js b/node_modules/bmaplib.lushu/index.js
index a793df6..0274623 100644
--- a/node_modules/bmaplib.lushu/index.js
+++ b/node_modules/bmaplib.lushu/index.js
@@ -352,6 +352,7 @@
         _moveNext: function(index) {
             var me = this;
             if (index < this._path.length - 1) {
+                me._opts.onTick && me._opts.onTick(index);
                 me._move(me._path[index], me._path[index + 1], me._tween.linear);
             } else {
                 me.stop()
